src/stridelayout.jl

```diff
@@ -38,6 +38,14 @@ function _offsets(x::X, dim::StaticInt{D}) where {X,D}
     end
 end
 
+"""
+    offset1(x) -> Integer
+
+Returns the offset of the linear indices for `x`.
+"""
+offset1(x) = _offset1(has_parent(x), x)
```

Comment:
> Regardless of the array's native indices, linear indices always range from 1:length(A). However, this raises an ambiguity for one-dimensional arrays (a.k.a., AbstractVector): does v[i] mean linear indexing , or Cartesian indexing with the array's native indices?

https://docs.julialang.org/en/v1/devdocs/offset-arrays/#Linear-indexing-(LinearIndices)

Maybe we should make the signature `offset1(::AbstractVector)`, define a generic `offset1(::AbstractArray) = static(1)`?

Reply:
But I guess other array types don't have to be beholden to this, since it is just the `OffsetArray`s docs.

Reply:
Originally I was trying to get this to the point where we could retrieve the value from a pointer and views of arrays might have different `offset1` values. We'll probably need a more elaborate definition in the future.